conn.go

```diff
@@ -84,6 +84,11 @@ func newConn(c engineio.Conn, handlers map[string]*namespaceHandler, broadcast B
 func (c *conn) Close() error {
 	var err error
 	c.closeOnce.Do(func() {
+		// For each namespace, leave all rooms, and call the disconnect handler.
+		for ns, nc := range c.namespaces {
+			nc.LeaveAll()
+			c.handlers[ns].onDisconnect(nc, "bye")
```

Comment:
Is there a potentional null pointer exception here?

Reply:
Yes, I believe so, check added.  Thank you for pointing this out.

Although, I do not fully understand the expectations of what namespaces should exist in the `handlers` and `namespaces` maps.  For instance, I see that `(*conn).connect` will always register a `namespaceConn` in `c.namespaces` for the root namespace, but there need not be a matching `namespaceHandler` in `handlers`.  Similarly, `(*conn).newConn` registers the provided `handlers`, but there need not be matching namespaces in `namespaces` as these are added on connect.

Given the above, it does seem like the right way to clean up is to cycle through the `namespaceConnection`s (in `c.namespaces`), triggering any registered disconnect handlers for those namespaces, but the intent is not well documented in conn.go, so please check that conclusion.